### PR TITLE
Add IDs to WebElements for integration testing (Ready for review)

### PR DESCRIPTION
**Pull request by jdreeve**
Adds an ID to any WebElement with a data-hook but no ID, with a value equivalent to the value of the data-hook.
Example:

WebElement with id="sample-id", data-hook="sample-hook": unchanged
WebElement with no id and data-hook="sample-hook": id="sample-hook" added
WebElement with no id and no data-hook: unchanged

This makes it easier for Selenium to consistently locate WebElements by ID for browser automation/integration testing.

### CODE REVIEW

File: client/templates/includes/editCustomStoichSpecie.pug [hunk 1]
```diff
@@ -1,10 +1,10 @@
 div
   div.inline
-    button.btn.btn-outline-secondary(class='btn-sm', data-hook='decrement') -
-  span.custom(data-hook="ratio")
+    button.btn.btn-outline-secondary(class='btn-sm', data-hook='decrement') -id="",
```
Comment (BryanRumsey) [inline client/templates/includes/editCustomStoichSpecie.pug:3]: This id attribute should be inside the parens, the '-' is the text for the button

File: client/templates/includes/editCustomStoichSpecie.pug [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+    button.btn.btn-outline-secondary(class='btn-sm', data-hook='decrement') -id="", 
+  span.custom(id="ratio", data-hook="ratio")
   div.inline
-    button.btn.btn-outline-secondary(class='btn-sm', data-hook='increment') +
-  select(data-hook="select-stoich-specie")
-    span.message.message-below.message-error(data-hook="message-container")
-      p(data-hook="message-text")
-  button.btn.btn-outline-secondary.custom(class='btn-sm', data-hook='remove') X
\ No newline at end of file
+    button.btn.btn-outline-secondary(class='btn-sm', data-hook='increment') +id="",
```
Comment (BryanRumsey) [inline client/templates/includes/editCustomStoichSpecie.pug:6]: This id attribute should be inside the parens, the '+' is the text for the button

**Comment by jdreeve** [comment]
Fixed dual/broken ID tags.

File: client/templates/includes/gillespyResults.pug [hunk 3]
```diff
@@ -18,21 +18,21 @@ div#workflow-results.card.card-body
...
-            td: div(data-hook="yaxis")
+            td: div(id="title", data-hook="title")
+            td: div(id="xaxis", data-hook="xaxis")
+            td: div(id="yaxis", data-hook="yaxis")
     div.card.card-body
       div
 
         h5.inline Plot Trajectories
-        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-trajectories", data-hook="collapse-trajectories") -
+        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-trajectories", id="collapse-trajectories", data-hook="collapse-trajectories") -
```
Comment (BryanRumsey) [inline client/templates/includes/gillespyResults.pug:28]: The element on line 30 already has this id value I would suggest adding '-btn' to the end of the existing id value.

No element use my suggested id value

File: client/templates/includes/gillespyResultsEnsemble.pug [hunk 4]
```diff
@@ -18,19 +18,19 @@ div#workflow-results-ensemble.card.card-body
...
+            td: div(id="title", data-hook="title")
+            td: div(id="xaxis", data-hook="xaxis")
+            td: div(id="yaxis", data-hook="yaxis")
 
     div.card.card-body
       div
 
         h5.inline Plot Standard Deviation Range
-        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-stddevrange", data-hook="collapse-stddevrange") -
+        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-stddevrange", id="collapse-stddevrange", data-hook="collapse-stddevrange") -
```
Comment (BryanRumsey) [inline client/templates/includes/gillespyResultsEnsemble.pug:29]: The element on line 31 already has this id value I would suggest adding '-btn' to the end of the existing id value.

No element use my suggested id value

File: client/templates/includes/gillespyResultsEnsemble.pug [hunk 5]
```diff
@@ -39,25 +39,25 @@ div#workflow-results-ensemble.card.card-body
       div
 
         h5.inline Plot Trajectories
-        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-trajectories", data-hook="collapse-trajectories") +
+        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-trajectories", id="collapse-trajectories", data-hook="collapse-trajectories") +
```
Comment (BryanRumsey) [inline client/templates/includes/gillespyResultsEnsemble.pug:42]: The element on line 44 already has this id value I would suggest adding '-btn' to the end of the existing id value.

No element use my suggested id value

File: client/templates/includes/gillespyResultsEnsemble.pug [hunk 6]
```diff
@@ -39,25 +39,25 @@ div#workflow-results-ensemble.card.card-body
...
+        button.btn.btn-primary.box-shadow(id="plot-multiple", data-hook="plot-multiple", disabled) Multiple Plots
         button.btn.btn-primary.box-shadow(id="trajectories" data-hook="download-png-custom" disabled) Download PNG
         button.btn.btn-primary.box-shadow(id="trajectories" data-hook="download-json" disabled) Download JSON
 
     div.card.card-body
       div
 
         h5.inline Plot Standard Deviation
-        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-stddev", data-hook="collapse-stddev") +
+        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-stddev", id="collapse-stddev", data-hook="collapse-stddev") +
```
Comment (BryanRumsey) [inline client/templates/includes/gillespyResultsEnsemble.pug:56]: The element on line 58 already has this id value I would suggest adding '-btn' to the end of the existing id value.

No element use my suggested id value

File: client/templates/includes/gillespyResultsEnsemble.pug [hunk 7]
```diff
@@ -66,13 +66,13 @@ div#workflow-results-ensemble.card.card-body
       div
 
         h5.inline Plot Trajectory Mean
-        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-trajmean", data-hook="collapse-trajmean") +
+        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-trajmean", id="collapse-trajmean", data-hook="collapse-trajmean") +
```
Comment (BryanRumsey) [inline client/templates/includes/gillespyResultsEnsemble.pug:69]: The element on line 61 already has this id value I would suggest adding '-btn' to the end of the existing id value.

No element use my suggested id value

File: client/templates/includes/parameterSweepResults.pug [hunk 8]
```diff
@@ -52,13 +52,13 @@ div#workflow-results.card.card-body
       div
 
         h5.inline Parameter Sweep
-        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-psweep", data-hook="collapse-psweep") -
+        button.btn.btn-outline-collapse(data-toggle="collapse", data-target="#collapse-psweep", id="collapse-psweep", data-hook="collapse-psweep") -
```
Comment (BryanRumsey) [inline client/templates/includes/parameterSweepResults.pug:55]: The element on line 57 already has this id value I would suggest adding '-btn' to the end of the existing id value.

No element use my suggested id value

**Pull request closed without merging**